### ISSUE TITLE
Doesn't auto-play next episode

### ISSUE THREAD

**Issue by nuentes** [im85288/service.upnext#199]
I've been having a weird issue with UpNext for a while. Hard to say how long, but I believe it goes back several versions. The dialog comes up just fine, and I can click both buttons, and they both work correctly if I manually click them. However, if I do nothing, the video simply ends. In the settings, I definitely have "Default action when nothing selected" correctly set to "Play".

I've tried an uninstall/reinstall of UpNext, and I've also downgraded from 1.1.2 to 1.0.5, and I have the exact same issue with either version. I was previously using the BOX skin, but I'm switching to Arctic Zephyr 2 and seeing the same issue on AZ2. All my media is added via Emby for Kodi. I have also confirmed that I do not have anything selected in the "Play next video automatically" setting for Kodi.

This logfile is from a clean Kodi boot (I still have 1.0.5 at the time of testing). I played an episode of a series. I skipped to the end of the episode and waited for the dialog to come up. For the first test episode, I selected "Watch Now" and the next episode correctly played. For the 2nd episode, I simply did nothing. The dialog came up, and the video player simply closed after the 20 seconds. Relevent part of my kodi logfile: https://paste.kodi.tv/vepepereca.kodi

**Comment by dagwieers** [comment]
This is probably a duplicate of #139 or #170, please test PR #198

**Comment by nuentes** [comment]
It looks like those commits were merged into master, so I downloaded and manually installed a zip of the master branch, and still had the same issue. I disabled and re-enabled the service just to be sure. I previously still had 1.0.5 installed (with auto-updates disabled) and now it shows 1.1.2 again.

I didn't mention it before, but this is installed on an Android device, if that's relevant.

**Comment by nuentes** [comment]
Based on updates in 198, it sounds like this is something that could potentially be resolved by the way an addon handles playback. Would it make sense to report this in the Emby forums?

**Comment by im85288** [comment]
@nuentes Emby for Kodi has two playback methods. If you selected Direct Paths (which I advise) then the behaviour is just like having your media locally installed in Kodi. The other method would use our addon integration method for showing upnext. It has all worked well in the past but not sure why it's not now. @angelblue05 does not seem to be active on Emby for Kodi so I think you'll have more success pursuing the issue here.

**Comment by nuentes** [comment]
Direct Paths requires me to change how my media is added in Emby. Right now my media is added via local folder paths. However, in the near future, I hope to move them to NFS or SMB paths, which would allow me to switch to Direct Play.

I currently only have Kodi installed on Android Devices. I'll install it on my laptop to test.

**Comment by im85288** [comment]
Can you do me a favour and test via Embycon if your not using direct paths. If it works correctly with Embycon we know it's Emby for Kodi that needs fixing.

**Comment by nuentes** [comment]
Identical issue with Embycon.

**Comment by GOvEy1nw** [comment]
I have the same issue here. 

I've just moved from Plex to Emby and have always used Kodi as the 'client' for either servers. When using Plex Kodi Connect (via it's default Add-on Path method) 'Up Next' worked pretty flawlessly, albeit with a kind of weird delay between episodes where it would pop back to the screen I started playing the episode from and hang there for a second or two, I guess while PKC prepped and loaded the next episode, but it did work all the time.

With Emby for Kodi (again using add-on path method), 'Up Next' rarely works (like it works maybe 5% of the time, but when it does work it goes straight to the next episode with no weird waiting on the selection screen etc. But again it doesn't work most of the time at all.

Also as a side note, I noticed that once when I was watching an episode of Show A, when then next up pop up appeared it was suggesting it was going to play an episode of Show B, even though there were still more episodes of show A to continue watching. So that was weird too.

I haven't tried Direct Path method as I'm aware this will reduce some usability between Kodi and Emby, so I'd prefer to keep with the Add-on path option if at all possible.

If you need any logs from me too, let me know and I'll sort some for you.

Cheers!

**Comment by nuentes** [comment]
> once when I was watching an episode of Show A, when then next up pop up appeared it was suggesting it was going to play an episode of Show B, even though there were still more episodes of show A to continue watching. So that was weird too.

Yeah, now that you mention it, I've seen this several times as well. For me, it usually only has come up when Show A had no more episodes.

**Comment by GOvEy1nw** [comment]
> > once when I was watching an episode of Show A, when then next up pop up appeared it was suggesting it was going to play an episode of Show B, even though there were still more episodes of show A to continue watching. So that was weird too.
> 
> Yeah, now that you mention it, I've seen this several times as well. For me, it usually only has come up when Show A had no more episodes.

I think that behaviour might make a little more sense, but in my case there were definitely more episodes left, & I had the 'include already watched episodes' option ticked or whatever it's called, so strange that it happened in my case.

**Comment by dagwieers** [comment]
> once when I was watching an episode of Show A, when then next up pop up appeared it was suggesting it was going to play an episode of Show B, even though there were still more episodes of show A to continue watching. So that was weird too.

This is a bug in the Up Next integration of the add-on. It is best to report this upstream as an issue in order to get this resolved. Up Next only processes what it receives, so in this case it appears to have received the wrong thing?

**Comment by im85288** [comment]
When I get the chance I'll also test. I used to use EmbyCon all the time and worked with the Dev to make sure everything worked. Recently though I use Emby for Kodi with Direct Paths (and as stated before this method simulates having a native Kodi database..which therefore means it works perfectly).

**Comment by dagwieers** [comment]
> I've just moved from Plex to Emby and have always used Kodi as the 'client' for either servers. When using Plex Kodi Connect (via it's default Add-on Path method) 'Up Next' worked pretty flawlessly, albeit with a kind of weird delay between episodes where it would pop back to the screen I started playing the episode from and hang there for a second or two, I guess while PKC prepped and loaded the next episode, but it did work all the time.

This is caused by the delay of triggering the add-on to play the next episode (and the time it takes to start playing). We could tigger it earlier (when we near the end of the video), but risk cutting off the video too soon. We do not know how much time it will take to start the next episode so Kodi simply returns to the calling menu.

Since v1.1.0 we support `play_url` playback, which makes Up Next perform playing the next episode (so avoids the call to the add-on service). This should help in a more efficient codepath.

In this master branch we add the upcoming episode to the playlist, which should make things smoother when the video finishes. However it still depends on the time it takes to start the next episode (possibly re-authenticating).


> With Emby for Kodi (again using add-on path method), 'Up Next' rarely works (like it works maybe 5% of the time, but when it does work it goes straight to the next episode with no weird waiting on the selection screen etc. But again it doesn't work most of the time at all.

This requires some troubleshooting to understand what is going on. There is a bug in Kodi related to concurrent busy dialogs in Kodi (possibly fixed in xbmc/xbmc#18104) that could be happening here too.

**Comment by nuentes** [comment]
I just started thinking, and it seems like it's related to what you're discussing above. As I said, clicking "Watch Now" works, but waiting for the auto-playback fails. I know this idea is a bandaid solution, but would it be possible to detect if the filepath looks like an emby path, and if it is, UpNext waits until there is a very tiny increment of video playback remaining (say 0.01s), and then it simply replicates the "Watch Now" action to play the next episode?

**Comment by nuentes** [comment]
I was able to get Direct Play configured today, and can confirm that this resolves the issue.

**Comment by im85288** [comment]
Thanks for confirming, I know with the addon paths you can use cinema intros..but direct paths is definitely the more robust option. It also assures your server is not transcoding and therefore using up resources.

**Comment by GOvEy1nw** [comment]
I don't mind using direct paths in my house, but I also share my library with family and with direct paths I expect it would be much more difficult to do remotely, plus I wouldn't then be able to see when people are watching something etc

**Comment by nuentes** [comment]
I also didn't really know how Direct paths worked until I configured them. Adding direct paths does not limit or change what you currently do, it simply adds another option. So I added the direct paths on the emby server, and configured Kodi to use the direct paths. However, I could have left Kodi the way it was, and it would have still functioned the same. Also, I did notice that Emby still tracked when I was watching something, and I could still see the status in the Dashboard.

**Comment by GOvEy1nw** [comment]
> So I added the direct paths on the emby server, and configured Kodi to use the direct paths. 

Great, ok I'll give it a shot, do you know of a guide anywhere to get it set up? As I'm using an Nvidia Shield and at the minute I've not managed to get Direct Paths working, even though the Shield itself can mount network storage. I assume the issue might be that I've not done anything about 'direct paths' on the emby server end of things. What settings do I have to set/change there?

**Comment by im85288** [comment]
Either you can edit the Library path or there’s an option in the library section for a network path.

As a rule never create your library pointing at for example the C: of a computer instead use UNC mapping.

For example: 192.168.1.100/tvshows 

I know the shield dose Samba sharing so you’ll be able to get it working fairly easily.

**Comment by GOvEy1nw** [comment]
> Either you can edit the Library path or there’s an option in the library section for a network path.
> 
> As a rule never create your library pointing at for example the C: of a computer instead use UNC mapping.
> 
> For example: 192.168.1.100/tvshows
> 
> I know the shield dose Samba sharing so you’ll be able to get it working fairly easily.

great thanks, I'm working on doing that now. Is there a way to do 'direct paths' for remote users too? Obviously 192.168.1.100 wont register on their end. I do have a ddns address for the nas etc, so could it work through that? Or is direct paths just not possible remotely?

**Comment by im85288** [comment]
Direct paths will work remotely..but the server will switch to transcoding for remote users if needed (bandwidth etc). I share LiveTV with relatives and I know that is marked as direct playing. Movies/TVShows are transcoded as my upload limit is 4mb.

**Comment by GOvEy1nw** [comment]
Ok, I'm stuck again... I've set the network paths up in emby as follows:

\\SERVERNAME\Media\MoviesFolder\

And I've even set up the NAS to connect to Kodi. I did this in Kodi by going to File Manager > Add Source > Browse > Add network Location, and added \\SERVERNAME as the location. I also proceeded to add it as an actual source by then clicking 'OK' on that new location & ok again to add it as a source.

I've also ensured that Kodi can access this by then going into the source in file manager, locating a movie and playing it directly.

However, when I try to set Emby for Kodi to direct play, go through all the initial yes/no options, but when it comes to syncing it tells me it can't find 'XYZ' movie, please make sure kodi can see the server etc etc... Even though the movie I played through the source I added was the exact same movie, so Kodi can clearly see it.

I'm not sure what I can be doing wrong.

**Comment by nuentes** [comment]
You might be better served in the emby forums. But when you added the network location in kodi, did you add the same exact server name? It likely would have shown up in there as smb:/SERVERNAME, or did you add by IP address? Is this an SMB share or NFS share? If NFS, are you sure the instance of Kodi has an IP that is allowed to access the server? If SMB, did you enter the credentials for the share when adding to the kodi file manager?

I believe as long as Kodi can traverse the network paths via file manager, then as long as the direct paths are using the same paths, it should work.

**Comment by im85288** [comment]
Hmm so it seems like your setup is near enough the same as mine, instead of network paths maybe set your main folders in Emby using the UNC paths, for example for tvshows I have:

`\\DISKSTATION\share\TV `

You do not need to add it as a source in Kodi. 

I think if you do it this way (ie as the main folder in Emby) then you guarantee the Emby Server can see it as if it cannot you’ll not have any media.

**Comment by GOvEy1nw** [comment]
Hey, I've just tried setting the main folder of one of my libraries to: `\\DISKSTATION\Plex Media\Kids Area\` (still got it named plex lol), but it just gave me an error saying 'The path could not be found. Please ensure the path is valid and try again.'

I've also tried it with the IP address instead and even tried it like `\\DISKSTATION\Plex/ Media\Kids/ Area/` in case the spaces in the folder names were causing issues, but it says the same thing oddly...

**Comment by im85288** [comment]
So you’ve got a Synology NAS the same as me, can you map to that folder on windows for example? Maybe this is a Shield server issue..though it doesn’t make sense.

I would expect you to have to use the share folder as in my path..so for example 

`to: \\DISKSTATION\share\Plex Media\Kids Area `

As all shared folders in synology are accessed that way.

**Comment by dagwieers** [comment]
Did anyone already tried if using the master branch improved the original issue? Since we now add it to the playlist we avoid the known bug in Kodi that could have been the cause of not playing the next episode.

**Comment by GOvEy1nw** [comment]
> So you’ve got a Synology NAS the same as me, can you map to that folder on windows for example? Maybe this is a Shield server issue..though it doesn’t make sense.
> 
> I would expect you to have to use the share folder as in my path..so for example
> 
> `\\DISKSTATION\share\Plex Media\Kids Area `
> 
> As all shared folders in synology are accessed that way.

Hey, yes, so I've even re-arranged all my data & folders (it was getting a little all over the place) and I've confirmed I can get to locations such as:

\\MyNas\Other_Media\Kids_Area\

Of course, at first it asks me to login, which I do, and then I can see and work with everything within that folder.

But if add that same folder to Emby (I've cleared all folders in all libraries at this point by the way) I just get the error 'The path could not be found. Please ensure the path is valid and try again.'...

I think it might be time to take this over to the Emby forums, I see that you're active over there too, hopefully we can all figure this out for the good of mankind haha.

**Comment by GOvEy1nw** [comment]
Just thinking.... Could the issue be that I'm running Emby from the NAS itself? So it can't see those folders as 'network' folders since to Emby it's all local? Seems daft if so, but just a thought...

**Comment by im85288** [comment]
@GOvEy1nw No idea why your having issues with it, agree probably best to take it to the Emby forums. I have found Emby to be pretty flakey with initial setup of shared folders sometimes...in particular it didn’t work well with the Emby server on Mac but was fine with the Emby server on Windows.

**Comment by nuentes** [comment]
@dagwieers Emby is basically the same idea as Plex. It's basically a back end that takes over media management responsibilities. Also, similarly to Plex, it can serve up videos either directly or with on-the-fly transcoding. It runs on a personal server. Media from Emby can get into Kodi 1 of 2 ways. The Emby for Kodi add-on, which basically adds nodes for TV/Movie/Music so the Kodi experience is exactly the same as if the media was local. The other addon option is Embycon. This one does not create local nodes, but rather connects directly to the server and queries the server directly any time you navigate. If you've used the Kodi Netflix add-on - it's kind of like that. But again, it's hitting your personal media on your personal server.

Any connection issues with Emby would present as an issue in other ways. UpNext wouldn't really be a suspect for Emby connection issues, except in possibly rare cases with impeccable timing.

**Comment by dagwieers** [comment]
Please test the upcoming v1.1.3 release by testing PR #217 and reporting back.

**Comment by diabl0w** [comment]
I have this issue as well, but I am not using any third party addon... I am playing the files natively and "locally" from an NFS share. Sometimes it does work, but only for very specific episodes of a season (and there is no difference in the codecs of the episodes, so that is not the cause. For example, letting the dialog box time out going from S01E03 -> S01E04 always works, but going from S01E04->S01E05 etc etc never works.
What happens with a working log:

> 2020-10-23 10:54:19.600 T:3753943936  NOTICE: [service.upnext] PlaybackManager -> played in a row settings 5
> 2020-10-23 10:54:19.604 T:3753943936  NOTICE: [service.upnext] PlaybackManager -> played in a row 1
> 2020-10-23 10:54:19.608 T:3753943936  NOTICE: [service.upnext] PlaybackManager -> showing next up page as played in a    row is 1
> 2020-10-23 10:54:42.325 T:3588305792  NOTICE: CVideoPlayer::OnExit()
> 2020-10-23 10:54:42.325 T:3588305792  NOTICE: VideoPlayer: eof, waiting for queues to empty
> 2020-10-23 10:54:42.325 T:3588305792  NOTICE: Closing stream player 1
> 2020-10-23 10:54:42.325 T:3588305792  NOTICE: CDVDMessageQueue(audio)::WaitUntilEmpty
> 2020-10-23 10:54:42.325 T:3588305792  NOTICE: Waiting for audio thread to exit
> 2020-10-23 10:54:42.325 T:3481904000   ERROR: Got MSGQ_ABORT or MSGO_IS_ERROR return true
> 2020-10-23 10:54:42.325 T:3481904000  NOTICE: thread end: CVideoPlayerAudio::OnExit()
> 2020-10-23 10:54:42.326 T:3588305792  NOTICE: Closing audio device
> 2020-10-23 10:54:42.709 T:3588305792  NOTICE: Deleting audio codec
> 2020-10-23 10:54:42.709 T:3588305792  NOTICE: Closing stream player 2
> 2020-10-23 10:54:42.709 T:3588305792  NOTICE: CDVDMessageQueue(video)::WaitUntilEmpty
> 2020-10-23 10:54:43.730 T:3588305792  NOTICE: waiting for video thread to exit
> 2020-10-23 10:54:43.730 T:3456734080   ERROR: Got MSGQ_ABORT or MSGO_IS_ERROR return true
> 2020-10-23 10:54:43.730 T:3456734080  NOTICE: thread end: video_thread
> 2020-10-23 10:54:43.730 T:3588305792  NOTICE: deleting video codec
> 2020-10-23 10:54:43.806 T:4059258896  NOTICE: CVideoPlayer::CloseFile()
> 2020-10-23 10:54:43.807 T:4059258896  NOTICE: VideoPlayer: waiting for threads to exit
> 2020-10-23 10:54:43.807 T:3753943936  NOTICE: [service.upnext] PlaybackManager -> playing media episode
> 2020-10-23 10:54:43.807 T:4059258896  NOTICE: VideoPlayer: finished waiting
> 2020-10-23 10:54:43.807 T:4059258896  NOTICE: CVideoPlayer::CloseFile()
> 2020-10-23 10:54:43.807 T:4059258896  NOTICE: VideoPlayer: waiting for threads to exit
> 2020-10-23 10:54:43.807 T:4059258896  NOTICE: VideoPlayer: finished waiting
> 2020-10-23 10:54:43.976 T:4059258896  NOTICE: VideoPlayer::OpenFile: nfs://192.168.111.1:2049/mnt/MyCloud/Media/TVShows/Survivorman/Season 1/S01E04.mkv
> 2020-10-23 10:54:43.977 T:3456734080  NOTICE: Creating InputStream
> 2020-10-23 10:54:44.006 T:3456734080  NOTICE: Creating Demuxer
> 2020-10-23 10:54:44.054 T:3456734080  NOTICE: Opening stream: 0 source: 256
> 2020-10-23 10:54:44.054 T:3456734080  NOTICE: CVideoPlayerVideo::OpenStream - Creating codec: 27
> 2020-10-23 10:54:44.054 T:3456734080  NOTICE: Creating video thread
> 2020-10-23 10:54:44.054 T:3481904000  NOTICE: running thread: video_thread
> 2020-10-23 10:54:44.055 T:3456734080  NOTICE: Opening stream: 1 source: 256
> 2020-10-23 10:54:44.055 T:3456734080  NOTICE: Finding audio codec for: 86019
> 2020-10-23 10:54:44.056 T:3456734080  NOTICE: CDVDAudioCodecFFmpeg::Open() Successful opened audio decoder ac3
> 2020-10-23 10:54:44.056 T:3456734080  NOTICE: Creating audio thread
> 2020-10-23 10:54:44.056 T:3319772032  NOTICE: running thread: CVideoPlayerAudio::Process()
> 2020-10-23 10:54:44.068 T:3319772032  NOTICE: Creating audio stream (codec id: 86019, channels: 6, sample rate: 48000, no pass-through)
> 2020-10-23 10:54:44.307 T:3753943936  NOTICE: [service.upnext] UpNextMonitor -> Up Next style autoplay succeeded

A not working log:

> 2020-10-23 10:51:58.305 T:3753943936  NOTICE: [service.upnext] PlaybackManager -> played in a row settings 5
> 2020-10-23 10:51:58.309 T:3753943936  NOTICE: [service.upnext] PlaybackManager -> played in a row 1
> 2020-10-23 10:51:58.313 T:3753943936  NOTICE: [service.upnext] PlaybackManager -> showing next up page as played in a row is 1
> 2020-10-23 10:52:21.742 T:3353342848  NOTICE: CVideoPlayer::OnExit()
> 2020-10-23 10:52:21.742 T:3353342848  NOTICE: VideoPlayer: eof, waiting for queues to empty
> 2020-10-23 10:52:21.742 T:3353342848  NOTICE: Closing stream player 1
> 2020-10-23 10:52:21.743 T:3353342848  NOTICE: CDVDMessageQueue(audio)::WaitUntilEmpty
> 2020-10-23 10:52:21.775 T:3353342848  NOTICE: Waiting for audio thread to exit
> 2020-10-23 10:52:21.776 T:3336557440   ERROR: Got MSGQ_ABORT or MSGO_IS_ERROR return true
> 2020-10-23 10:52:21.776 T:3336557440  NOTICE: thread end: CVideoPlayerAudio::OnExit()
> 2020-10-23 10:52:21.776 T:3353342848  NOTICE: Closing audio device
> 2020-10-23 10:52:21.934 T:3753943936  NOTICE: [service.upnext] PlaybackManager -> playing media episode
> 2020-10-23 10:52:22.008 T:4059258896  NOTICE: VideoPlayer::OpenFile: nfs://192.168.111.1:2049/mnt/MyCloud/Media/TVShows/Survivorman/Season 1/S01E05.mkv
> 2020-10-23 10:52:22.014 T:3753943936  NOTICE: [service.upnext] UpNextMonitor -> Up Next style autoplay succeeded
> 2020-10-23 10:52:22.226 T:3353342848  NOTICE: Deleting audio codec
> 2020-10-23 10:52:22.226 T:3353342848  NOTICE: Closing stream player 2
> 2020-10-23 10:52:22.226 T:3353342848  NOTICE: CDVDMessageQueue(video)::WaitUntilEmpty
> 2020-10-23 10:52:22.815 T:3353342848  NOTICE: waiting for video thread to exit
> 2020-10-23 10:52:22.815 T:3344950144   ERROR: Got MSGQ_ABORT or MSGO_IS_ERROR return true
> 2020-10-23 10:52:22.815 T:3344950144  NOTICE: thread end: video_thread
> 2020-10-23 10:52:22.815 T:3353342848  NOTICE: deleting video codec
> 2020-10-23 10:52:22.877 T:4059258896  NOTICE: CVideoPlayer::CloseFile()
> 2020-10-23 10:52:22.878 T:4059258896  NOTICE: VideoPlayer: waiting for threads to exit
> 2020-10-23 10:52:22.878 T:4059258896  NOTICE: VideoPlayer: finished waiting
> 2020-10-23 10:52:22.878 T:4059258896  NOTICE: CVideoPlayer::CloseFile()
> 2020-10-23 10:52:22.878 T:4059258896  NOTICE: VideoPlayer: waiting for threads to exit
> 2020-10-23 10:52:22.878 T:4059258896  NOTICE: VideoPlayer: finished waiting

As you can see, in the working version, up next doesnt try to open the new file until after "finished waiting", but it seems like in the failed version it tries to open the file before kodi is finished closing the previous video

**Comment by diabl0w** [comment]
And for the good news, I git pulled the current master branch and the issue has been fixed!

**Comment by cjk28** [comment]
Confirmed fixed for me too. Potentially not directly related, if I jump to the end of an episode by skipping (e.g. pressing up on the remote to skip to the end), the add-on will not be invoked and therefore the next episode won't be played (unless we are already in a playlist, which for my use case of generally launching a show from the "TV shows -next episodes"widget is not the case. Could that be handled by the add-on?

**Comment by dagwieers** [comment]
We planned to do a new release this weekend. I did not make it, but it will be there ASAP.

> Could that be handled by the add-on?

Please open a new issue for this.

**Comment by popy2k14** [comment]
Any news for this issue.
This also doesn't autoplay for me on the end of an episode. When I press the next button, it works.
Thx

**Comment by MoojMidge** [comment]
@popy2k14 What version are you using?

**Comment by popy2k14** [comment]
Newest v1.1.4 on Leia v18.9 with emby for kodi.

**Comment by MoojMidge** [comment]
Can you provide a debug log? The fixes discussed here either involved changing the way Emby was integrated with Kodi or using a playlist generated by UpNext to queue and play the next episode.

Either of these fixes should work for the majority of cases, however there were additional edge cases where one or more of the following could occur:
* addon data was received after onPlaybackStarted had already been called
* incorrect callbacks were being called when there was a playback error
* expected order of callbacks being called in UpNext logic was not consistent with actual order of callbacks being called by Kodi
* video stream info had not fully loaded when onPlaybackStarted or onAVStarted had been called resulting in UpNext thinking that the stream duration was zero

Need more information to determine exactly what the problem is in your case.

**Comment by cjk28** [comment]
This seems to have returned for me after updating to Kodi 19.0. I have 1.1.5 of the addon.
log with upnext debug enabled and just opened kodi, played one episode, let the timer expire, expected next episode, but playback ends. jellyfin for kodi seems a bit unhappy at a glance of the log so maybe that is part of the problem
[kodi.log](https://github.com/im85288/service.upnext/files/6045919/kodi.log)

**Comment by MoojMidge** [comment]
For some reason your played in a row setting has been set to zero, so the Still Watching? popup is always showing, not the Up Next popup.

If there is no user interaction on the Still Watching? popup then playback will stop.

You need to change the setting for `Number of episodes before "Still there?" query` to something other than zero

**Comment by cjk28** [comment]
Thanks for the quick reply and analysis! that has indeed fixed it, although i hadn't changed my settings, so I believe i had it set to 0 before and it was working, despite definitely seeing the "still watching?" popup.

Either way the settings/behaviour on this topic are a bit unclear, it would be good if the settings page could be reworked a bit to make this more obvious.
thanks again!

**Comment by MoojMidge** [comment]
Yeah that was one of the issues that was also fixed in v1.1.5

I think there are a few opportunities to improve the Still Watching? functionality:
- currently the comparison of played episodes is made to played_in_a_row_number + 1 (effectively, the actual comparison is played_in_a_row <= played_in_a_row_number) which means that a user would set the `Number of episodes before "Still there?" query` setting to 3, for example, but the Still Watching? popup would only show up if 4 episodes have played and the 5th episode was about to play. This doesn't seem correct to me.
- There is no way to disable the Still Watching? popup. Setting to zero should disable it. Setting to one should mean it appears all the time. 
- Tidy up wording and add more description.
